File: de.lmu.ifi.dbs.medmon.datamining.core/src/de/lmu/ifi/dbs/medmon/datamining/core/filter/BypassFilter.java
package de.lmu.ifi.dbs.medmon.datamining.core.filter;

import de.lmu.ifi.dbs.medmon.datamining.core.container.RawData;
import de.lmu.ifi.dbs.medmon.datamining.core.processing.AbstractDataProcessor;
import de.lmu.ifi.dbs.medmon.datamining.core.processing.IFilter;

/**
 * This filter just forwards the input
 * Just for testing issues
 * @author Nepomuk Seiler
 *
 */
public class BypassFilter extends AbstractDataProcessor implements IFilter {

	public BypassFilter() {
		super("Bypass Filter", INDEFINITE_DIMENSION, INDEFINITE_DIMENSION);
	}

	@Override
	public RawData process(RawData data) {
		return data;
	}


}

File: knowing.test/src/knowing/test/filter/BypassFilter.scala
package knowing.test.filter

import akka.event.EventHandler.{debug, info, warning, error}
import de.lmu.ifi.dbs.knowing.core.processing.TFilter
import de.lmu.ifi.dbs.knowing.core.factory._
import de.lmu.ifi.dbs.knowing.core.util.ResultsUtil
import weka.core.{ Instances, Instance }
import java.util.Properties
import akka.actor.ActorRef
import akka.actor.Actor

/**
 * Simple bypass filter.
 * 
 * @author Nepomuk Seiler
 * @since 2011-08-22
 * @version 1.0
 */
class BypassFilter extends TFilter {

  def filter(instances: Instances): Instances =  instances
  
  def configure(properties: Properties) {}

  def query(query: Instance): Instances = {
    val ret = new Instances(query.dataset, 0)
    ret.add(query)
    ret
  }

  def result(result: Instances, query: Instance) { }
}

class BypassFilterFactory extends ProcessorFactory(classOf[BypassFilter]) 
